spec/game_spec.rb

```diff
@@ -0,0 +1,215 @@
+      expect(game.chose_difficulty('exit')).to eq(:exit)
+    end
+
+    it "returns false if bad user's input" do
+      expect(game.chose_difficulty('unknown_command')).to eq(false)
+    end
+  end
+
+  context '#initialize' do
+    it 'should generate array with code' do
+      expect(game.instance_variable_get(:@code).is_a?(Array)).to eq(true)
+    end
+    it 'should have integers in array' do
+      expect(game.instance_variable_get(:@code).map { |element| element.is_a? Integer }).to eq([true, true, true, true])
+    end
+
+    it 'should create array with length 4' do
+      expect(game.instance_variable_get(:@code).length).to eq(4)
+    end
+
+    let(:arr) { game.instance_variable_get(:@code) }
+
+    it 'should have numbers between 1 and 6' do
+      expect(arr[0]).to be_between(1, 6)
+      expect(arr[1]).to be_between(1, 6)
+      expect(arr[2]).to be_between(1, 6)
+      expect(arr[3]).to be_between(1, 6)
+    end
+  end
+
+  context '#user_guess' do
+    it 'returns hint symbol' do
+      expect(game.user_guess('hint')).to eq(:hint)
+    end
+
+    it 'returns rules symbol' do
+      expect(game.user_guess('rules')).to eq(:rules)
+    end
+
+    it 'returns exit symbol' do
+      expect(game.user_guess('exit')).to eq(:exit)
+    end
+
+    it 'returns false if code is a string' do
+      expect(game.user_guess('sdfs')).to eq(false)
+    end
+
+    it 'returns false if code is too short' do
+      expect(game.user_guess(12)).to eq(false)
+    end
+
+    it 'returns false if code is too long' do
+      expect(game.user_guess(1_234_567)).to eq(false)
+    end
+  end
+
+  context '#take_hint' do
+    let(:user) { User.new('User1', 1) }
+    let(:code) { game.instance_variable_get(:@code) }
+    let(:used_hints) { [] }
+
+    it 'returns integer' do
+      expect(game.take_hint(user, used_hints).class).to eq(Integer)
+    end
+
+    it 'should not change the code' do
+      old_code = code
+      game.take_hint(user, used_hints)
+      expect(code).to eq(old_code)
+    end
+
+    it 'returns one number from code' do
+      expect(code.include?(game.take_hint(user, used_hints))).to eq(true)
+    end
+
+    it 'returns different numbers after 2 hints' do
+      hint = game.take_hint(user, used_hints)
+      used_hints.push(hint)
+      expect(game.take_hint(user, used_hints)).not_to eq(hint)
+    end
+
+    it 'increases used hints at user' do
+      hints_before_test = user.hints_used
+      game.take_hint(user, used_hints)
+      expect(user.hints_used).to be > hints_before_test
+    end
+
+    it 'returns false if user have used all hints' do
+      game.take_hint(user, used_hints)
+      game.take_hint(user, used_hints)
+      expect(game.take_hint(user, used_hints)).to eq(false)
+    end
+  end
+
+  context '#compare_codes' do
+    before { game.instance_variable_set(:@code, [1, 3, 5, 6]) }
+    it 'returns array' do
+      user_code = [1, 2, 3, 4]
+      expect(game.compare_codes(user_code).class).to eq(Array)
```

Comment:
move `game.compare_codes(user_code)` to `let`

Reply:
fixed